### ISSUE TITLE
[Feature Request] add a clear database button

### ISSUE THREAD

**Issue by xdeiimoss** [AriaMoradi/Tachidesk#1199]
## What feature should be added to Suwayomi?
Currently, Suwayomi has only a button to remove non library manga from categories. There should be one that works just like the one from Mihon

**Comment by Robonau** [comment]
that button exists to fix an issue specific to suwayomi. 
there is no mihon counterpart. 
if you want to clear the database you can just delete the data files

**Comment by xdeiimoss** [comment]
> that button exists to fix an issue specific to suwayomi. there is no mihon counterpart. if you want to clear the database you can just delete the data files

There is a Mihon counterpart actually
It clears the database of entries that are not saved in the library 
![Screenshot_20241212_091429_Mihon](https://github.com/user-attachments/assets/0f649ee8-6743-49ba-80b1-e4cf94a263fc)

**Comment by Robonau** [comment]
suwayomi doesn't have history.
a button that "deletes history for entries not saved in library" on suwayomi would do nothing

**Comment by xdeiimoss** [comment]
It doesn't delete only the history though, but everything that isn't saved on the library, even if you haven't read it

**Comment by AvvAdeh** [comment]
I agree, I find weird that the database contains entries that we didn't even search for.

I use dbeaver and select all entries that aren't 'in library', it often cut the database size by more than 2 times.

database.mv.db
path: /PUBLIC/Tables/MANGA/
column: IN_LIBRARY
type: boolean

**Comment by AvvAdeh** [comment]
Maybe, instead of a clear database button for entries not in library...
Wouldn't it be possible by default to not save in the library entries that aren't in it?
If so, without a clear database button, that would imply to remove the entry in the database when removing an entry in the library.

**Comment by Robonau** [comment]
most of the entries in yr database were never in yr library

**Comment by xdeiimoss** [comment]
> Maybe, instead of a clear database button for entries not in library... Wouldn't it be possible by default to not save in the library entries that aren't in it? If so, without a clear database button, that would imply to remove the entry in the database when removing an entry in the library.

Most of the stuff you get in your database is a result of searching or even simply opening up any source. Removing only the entries that were in the library before from the database would literally make no difference comparing to an actual clear database button like it does in Mihon. There is a fork of TachiSY (Komikku) that made it so the entries only gets added to the database when the user opens them up, so no one has a bunch of redundant stuff in their database. Maybe this could be added here as well?

**Comment by Syer10** [comment]
There is no plans to change how Suwayomi caches manga in the database

**Comment by FancyBEAST** [comment]
+1 to clear database button like mihon
